compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -509,6 +517,64 @@ encodeModuleRule =
         let (hash, bs) = SS.encodeModule lfVersion m
         return ([], Just (mconcat $ hash : map fst encodedDeps, bs))
 
+-- hlint
+
+-- We need to provide hlint the path to the directory of a file
+-- containing a base configuration file called 'hlint.yaml'.
+-- We test with one version of 'hlint.yaml' and deploy with another
+-- (at the current time, there are good reasons for this). The code
+-- below (which relies on the behavior of `package_app` deploying to
+-- '~/.daml/sdk/x.x.x/damlc/resources/daml-ide-core/data') selects one
+-- or the other depending on context.
+getHlintDataDir :: IO FilePath
+getHlintDataDir = do
+  root <- locateRunfiles $
+    mainWorkspace </> "compiler/damlc/daml-ide-core"
+  let test = root </> "data/test"
+      prod = root </> "daml-ide-core/data/prod"
+  useProd <- System.Directory.doesDirectoryExist prod
+  return $ if useProd then prod else test
+
+-- This next function harvests hlint settings using 'hlint.yaml'. What
+-- we don't have at this time but I will follow up with in a later
+-- update, is the ability to augment the settings via per-project
+-- overrides from user provided '.hlint.yaml' files (found by by
+-- searching the directory from which 'daml studio' is invoked [and
+-- possibly its ancestors]). It's in anticpation of users updating
+-- these '.hlint.yaml' files that we read settings on every
+-- invocation.
+hlintSettings :: IO (ParseFlags, [Classify], Hint)
+hlintSettings = do
```

Comment:
Do you have an idea of how expensive this is? We probably want to cache this based on whether the config files have changed.

Reply:
I can do that. I don't think it's expensive but let's just do it right in any case. I'll add a rule. Shouldn't be too hard.

Reply:
Once you take user configs into account it gets a bit tricky but yeah for now, caching it should be easy.